plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1768,13 +1767,15 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             Map<IcebergColumnHandle, Domain> newUnenforced = new LinkedHashMap<>();
             Map<IcebergColumnHandle, Domain> domains = predicate.getDomains().orElseThrow(() -> new VerifyException("No domains"));
             domains.forEach((columnHandle, domain) -> {
-                // Iceberg metadata columns can not be used to filter a table scan in Iceberg library
                 // TODO (https://github.com/trinodb/trino/issues/8759) structural types cannot be used to filter a table scan in Iceberg library.
-                if (isMetadataColumnId(columnHandle.getId()) || isStructuralType(columnHandle.getType()) ||
+                if (isStructuralType(columnHandle.getType()) ||
                         // Iceberg orders UUID values differently than Trino (perhaps due to https://bugs.openjdk.org/browse/JDK-7025832), so allow only IS NULL / IS NOT NULL checks
                         (columnHandle.getType() == UUID && !(domain.isOnlyNull() || domain.getValues().isAll()))) {
                     unsupported.put(columnHandle, domain);
                 }
+                else if (columnHandle.isPathColumn()) {
```

Comment:
When dealing with a metadata column which is not  `$path`,  the column handle should land into `unsupported` map. Currently this check doesn't have any outcome (only `$path` metadata column is currently exposed).

Reply:
Write a test for `io.trino.plugin.iceberg.IcebergColumnHandle#TRINO_UPDATE_ROW_ID_COLUMN_NAME`/ `$row_id` to verify that the filter is being ignored in such case.

Reply:
`$row_id` isn't metadata column. I wouldn't handle the comment in this PR.

Reply:
`IcebergMetadataColumn` has only `$path` at this time. I modified like below anyway. 
```java
                else if (isMetadataColumnId(columnHandle.getId())) {
                    if (columnHandle.isPathColumn()) {
                        newEnforced.put(columnHandle, domain);
                    }
                    else {
                        unsupported.put(columnHandle, domain);
                    }
                }

```